test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/CompareAttributeAdapterTest.cs

```diff
@@ -138,15 +184,61 @@ public void ClientRulesWithCompareAttribute_ErrorMessageUsesResourceOverride()
             };
             var adapter = new CompareAttributeAdapter(attribute, stringLocalizer: null);
 
+            var expectedMessage = "Comparing MyProperty to OtherProperty.";
+
             var actionContext = new ActionContext();
-            var context = new ClientModelValidationContext(actionContext, metadata, metadataProvider);
+            var context = new ClientModelValidationContext(
+                actionContext,
+                metadata,
+                metadataProvider,
+                new AttributeDictionary());
+
+            // Act
+            adapter.AddValidation(context);
+
+            // Assert
+            Assert.Collection(
+                context.Attributes,
+                kvp => { Assert.Equal("data-val", kvp.Key); Assert.Equal("true", kvp.Value); },
+                kvp => { Assert.Equal("data-val-equalto", kvp.Key); Assert.Equal(expectedMessage, kvp.Value); },
+                kvp =>
+                {
+                    Assert.Equal("data-val-equalto-other", kvp.Key);
+                    Assert.Equal(kvp.Value, "*.OtherProperty");
+                });
+        }
+
+        [Fact]
+        [ReplaceCulture]
+        public void AddValidation_DoesNotTrounceExistingAttributes()
```

Comment:
These are great additions and they go beyond my original request in that you've touched all the adapters. However I don't see anything confirming unrelated attributes are left alone. Just need one or two of the adapter tests to check `attribute-name = "attribute-value"`, not all.